contents/performance/PerformanceProfiling.md

```diff
@@ -0,0 +1,132 @@
+# Performance Profiling
+
+Author: [Ong Heng Le](https://github.com/initialshl)
+
+## Overview
+
+Performance profiling involves collecting data (such as function timings) 
+about a program to identify areas with performance issues. It is a tool that 
+is usually used in code optimization. CPU sampling and instrumentation are 
+the two most common types of performance profiling methods.
+
+* **CPU sampling**: Collects *samples* at fixed intervals, which provides an overview of your program's performance
+* **Instrumentation**: Collects detailed *elapsed timings* about your program
+
+> If you are interested, here are [more technical details](https://blogs.msdn.microsoft.com/ejarvi/2005/04/07/the-choice-between-sampling-and-instrumentation/) about 
+[CPU sampling](https://msdn.microsoft.com/en-us/library/dd264994.aspx#Anchor_0) and 
+[instrumentation](https://msdn.microsoft.com/en-us/library/dd264994.aspx#Anchor_1).
+
+### Resources:
+1. [Profiling Overview](https://msdn.microsoft.com/en-us/library/bb384493(v=vs.110).aspx)
+1. [The Choice Between Sampling and Instrumentation](https://blogs.msdn.microsoft.com/ejarvi/2005/04/07/the-choice-between-sampling-and-instrumentation/)
+1. [Understanding Performance Collection Methods](https://msdn.microsoft.com/en-us/library/dd264994.aspx)
+
+# Performance Profiling For The First Time
+
+Most profilers are similar in their functionalities, user interface, and use of 
+technical terms. This tutorial demonstrates using the profiler included in 
+[Visual Studio 2015](https://www.visualstudio.com/downloads/).
+
+> You are encouraged to adapt this tutorial to your preferred profiling 
+tools on your own project. <br>
+
+## 1. Run a performance profiling session
+
+Before you start, you will be asked to choose a profiling method. It is recommended to 
+use **CPU sampling** for your first profiling session.
```

Comment:
Just checking, do you mean that we should use this every time we start a new session, or are you suggesting this as a beginner choice?